packages/core/src/modules/routing/services/MediationRecipientService.ts

```diff
@@ -104,6 +105,14 @@ export class MediationRecipientService {
     // Update record
     mediationRecord.endpoint = messageContext.message.endpoint
 
+    // Update connection metadata to use their key format in further protocol messages
+    const connectionUsesDidKey = messageContext.message.routingKeys.some(isDidKey)
+
+    const useDidKeysForProtocol = connection.metadata.get(ConnectionMetadataKeys.UseDidKeysForProtocol) ?? {}
+    useDidKeysForProtocol[MediationGrantMessage.type.protocolUri] = connectionUsesDidKey
```

Comment:
How does this work with minor version updates?

Reply:
This is tied to a specific protocol major.minor version. If a new interaction is started with a newer protocol version, a new key/value will be added. Not super nice in terms of storage optimization, but I guess this will be mostly limited to specific connections (such as mediators), so should not represent a big issue 🤔

Reply:
Ok, something to look out for. We maybe don't need to store it separtely per minor version (however if ACA-Py updated to mediator coordinator 1.1 and 1.0 already supported did:key I would assume the newer version also does).